components/app_bar/app_bar_plugin_component.tsx

```diff
@@ -42,38 +44,28 @@ const AppBarPluginComponent = (props: PluginComponentProps) => {
         return latestEntry;
     };
 
-    const getIconForPluginComponent = (component: PluginComponent): React.ReactNode => {
-        const entry = getMarketplaceEntryForPlugin(component.pluginId);
-        if (entry) {
-            return (
-                <div className={'app-bar__icon-inner'}>
-                    <img src={entry.icon_data} />
-                </div>
-            );
-        }
-
-        return component.icon;
-    };
-
-    const getPluginDisplayName = (component: PluginComponent): React.ReactNode => {
-        const text = component.tooltipText || component.text;
-        if (text) {
-            return text;
-        }
+    const entry = getMarketplaceEntryForPlugin(component.pluginId);
 
-        const entry = getMarketplaceEntryForPlugin(component.pluginId);
-        if (entry) {
-            return entry.manifest.name;
-        }
+    let displayName = component.pluginId;
+    if (entry) {
+        displayName = entry.manifest.name;
+    }
 
-        return component.pluginId;
-    };
+    let icon = component.icon;
+    if (entry) {
+        icon = (
+            <div className={'app-bar__icon-inner'}>
+                <img src={entry.icon_data} />
+            </div>
+        );
+    }
```

Comment:
```suggestion
    const icon = entry ? (
        <div className={'app-bar__icon-inner'}>
            <img src={entry.icon_data} />
        </div>
    ) : component.icon;
```

Reply:
Marketplace icon logic has been removed so this is not relevant in this PR anymore